app/src/main/java/com/google/droidjump/models/Droid.java

```diff
@@ -34,21 +37,19 @@
     // the variable is responsible for how much time ticks must pass to change a bitmap so that animate droid smoothly
     public final static int animationStepTicks = 2;
 
-
-    public Droid(int x, int y, Resources resources) {
-        super(x, y);
-        Bitmap fullDroidPicture = BitmapFactory.decodeResource(resources, R.mipmap.droid);
+    public Droid (int x, int y, Resources resources){
+        super(x, y, R.mipmap.droid, resources);
         int droidCount = GameConstants.DROID_COUNT_ON_FULL_DROID_PICTURE;
         droidTypes = new Bitmap[droidCount];
-        int droidWidth = fullDroidPicture.getWidth() / droidCount;
-        int droidHeight = fullDroidPicture.getHeight();
+        int droidWidth = picture.getWidth() / droidCount;
+        int droidHeight = picture.getHeight();
         for (int i = 0; i < droidCount; i++) {
-            droidTypes[i] = Bitmap.createBitmap(fullDroidPicture, /* x= */ droidWidth * i,
+            droidTypes[i] = Bitmap.createBitmap(picture, /* x= */ droidWidth * i,
                     /* y= */ 0, droidWidth, droidHeight);
         }
-        setBitmap(droidTypes[GameConstants.DROID_FIRST_STEP_INDEX]);
-        this.setY(y - getBitmap().getHeight());
-        initialY = getY();
+        picture = (droidTypes[GameConstants.DROID_FIRST_STEP_INDEX]);
```

Comment:
I am a bit confused. on line 44,45 you set the height and width of the droid, but you assign the droid on this line. Shouldn't it be at the begining of the method?

Reply:
I don't quite understand what exactly you mean by "assign"?

Reply:
this.y = (y - picture.getHeight()); I was referring to this line.

Reply:
Tried to resolve, is it looking ok now?

Reply:
Laura's concern is that you use "picture.getHeight()" on line 46 before variable "picture" has a value which it only gets on line 51. 

The reason why it works is that "super(x, y, R.mipmap.droid, resources);" actually assigns the value "R.mipmap.droid" to "picture" variable in the GameItem class constructor. To make it more clear, it probably would be better to move "this.y = (y - picture.getHeight());" after you give "picture" a value here.

Reply:
Thank you! Now I understand

Reply:
Done